post/webnote-extension/index.md

```diff
@@ -0,0 +1,658 @@
+-   Click on `Load Unpacked button` present at top left
+-   select project directory/dist folder
+
+![Load unpacked extension](./images/load-unpacked.png)
+
+![Extension Preview](./images/extension-preview.png)
+
+## Setup Routes
+
+-   Home page
+    contains list of notes added with url
+
+-   login page
+-   signup page
+-   Create and edit note page
+
+### Setup Signup
+
+Create new file `src/views/signup.vue`
+
+```html
+<template>
+    <div>
+        <form @submit.prevent="signup">
+            <label>Name</label>
+            <input type="text" v-model="name" required />
+
+            <label>Email</label>
+            <input type="email" v-model="email" required />
+
+            <label>Password</label>
+            <input type="password" v-model="password" required />
+            <button type="submit" style="margin-top: 16px">Signup</button>
+            <p>
+                Already have account?
+                <span
+                    @click="$router.push({ name: 'login' })"
+                    style="text-decoration: underline; cursor: pointer"
+                    >Login</span
+                >
+            </p>
+        </form>
+    </div>
+</template>
+
+<script>
+    import { createUser } from "@directus/sdk";
+    export default {
+        inject: ["directus"],
+        data() {
+            return {
+                name: "",
+                email: "",
+                password: "",
+            };
+        },
+        methods: {
+            async signup() {
+                const result = await this.directus.request(
+                    createUser({
+                        first_name: this.name,
+                        last_name: "",
+                        email: this.email,
+                        password: this.password,
+                        role: "1c005600-202d-429c-81bb-452a70aec7e1", // Customer
+                    })
+                );
+                this.$router.push({ name: "login" });
+            },
+        },
+    };
+</script>
+```
+
+Add css for button
+
+```css
+button {
+    border-radius: 8px;
+    border: 1px solid transparent;
+    padding: 0.6em 1.2em;
+    font-size: 1em;
+    font-weight: 500;
+    font-family: inherit;
+    background-color: #1a1a1a;
+    cursor: pointer;
+    transition: border-color 0.25s;
+}
+button:hover {
+    border-color: #646cff;
+}
+button:focus,
+button:focus-visible {
+    outline: 4px auto -webkit-focus-ring-color;
+}
+```
+
+Role: Its the `customer` role we created. To check, check the url in `https://directus-supabase.onrender.com/admin/settings/roles/1c005600-202d-429c-81bb-452a70aec7e1`
+
+### Load Signup route
+
+-   Open `src/router.js`
+-   Import `signup.vue`
+
+```js
+import SignupView from "../views/signup.vue";
+```
+
+Add the route in `routes`
+
+```js
+   {
+        path: "/signup",
+        name: "signup",
+        meta: { public: true },
+        component: SignupView,
+    },
+```
+
+![Signup](./images/signup.png)
```

Comment:
````suggestion
### Register Users

Create new file `src/views/signup.vue`

```html
<template>
    <div>
        <form @submit.prevent="signup">
            <label>Name</label>
            <input type="text" v-model="name" required />

            <label>Email</label>
            <input type="email" v-model="email" required />

            <label>Password</label>
            <input type="password" v-model="password" required />
            <button type="submit">Signup</button>
            <p>
                Already have account?
                <span @click="$router.push({ name: 'login' })">Login</span>
            </p>
        </form>
    </div>
</template>

<script>
    import { createUser } from "@directus/sdk";
    export default {
        inject: ["directus"],
        data() {
            return {
                name: "",
                email: "",
                password: "",
            };
        },
        methods: {
            async signup() {
                const result = await this.directus.request(
                    createUser({
                        first_name: this.name,
                        last_name: "",
                        email: this.email,
                        password: this.password,
                        role: "1c005600-202d-429c-81bb-452a70aec7e1", // Customer
                    })
                );
                this.$router.push({ name: "login" });
            },
        },
    };
</script>
```

Open `src/router.js` and import `signup.vue`:

```js
import SignupView from "../views/signup.vue";
```

Add the route in `routes`:

```js
{
    path: "/signup",
    name: "signup",
    meta: { public: true },
    component: SignupView,
},
```

![Signup](./images/signup.png)
````

Please remove all styling instruction from this post, including inline styling. Please also use the new registration endpoint (note in the Directus setup section to enable the setting). https://docs.directus.io/reference/authentication.html#register